### PR TITLE
Updated vrsettings logic to use new section suffix

### PR DESCRIPTION
**Pull request by noob247365**
The code has will fall back to the non-prefixed version if the new section is not found, for backwards compatibility. Writing however, will only ever write to the new version.

This is to be paired with LucidVR/opengloves-driver#130 to address LucidVR/opengloves-driver#129

### CODE REVIEW

File: src/utils/sidecar.js [hunk 1]
```diff
@@ -4,8 +4,19 @@ export const openSidecar = async (program, type, data) => {
...
+        const stderr = [];
+        sidecar.stderr.on('data', e => stderr.push(e));
+
+        // Report output on exit
+        sidecar.on('close', () => {
+            if (stderr.length > 0)
+                reject(stderr.join('\n'));
+            else
+                resolve(stdout.join('\n'));
+        });
```
Comment (danwillm) [inline src/utils/sidecar.js:19]: Good idea. Though I'd resolve/reject with the array instead of making it a string

**Review by danwillm — CHANGES_REQUESTED** [review]

File: src/utils/sidecar.js (hunk 1 above)
Comment (noob247365) [reply src/utils/sidecar.js]: Should be fixed with latest commit

File: src/utils/configuration.js [hunk 2]
```diff
@@ -6,12 +6,14 @@ import {deepOverwrite} from "./object";
...
 
+export const primaryConfigurationKey = 'opengloves.driver_openglove';
+
 /***
  * Returns values for configuration properties provided using external executable
  * @param configObj {Object} Object for properties
  * @return {Promise<String>} Return values from properties provided
  */
-const getValuesForConfiguration = async (configObj) => openSidecar("sidecar", "settings_get", configObj);
+const getValuesForConfiguration = async (configObj) => (await openSidecar("sidecar", "settings_get", configObj)).join('\n');
```
Comment (danwillm) [inline src/utils/configuration.js:16]:
```suggestion
const getValuesForConfiguration = async (configObj) => (await openSidecar("sidecar", "settings_get", configObj)).pop();
```

File: src/utils/sidecar.js [hunk 3]
```diff
@@ -4,8 +4,21 @@ export const openSidecar = async (program, type, data) => {
...
-        sidecar.stderr.on('data', e => reject(e));
+        // Listen for output
+        const stdout = [];
+        sidecar.stdout.on('data', e => stdout.push(e));
+        let hadStderr = false;
+        sidecar.stderr.on('data', e => {
+            if (!hadStderr)
+                reject(e);
+            hadStderr = true;
+        });
```
Comment (danwillm) [inline src/utils/sidecar.js:15]: I don't think the flag is neeeded, just return after rejecting.

**Review by danwillm — CHANGES_REQUESTED** [review]

File: src/utils/sidecar.js (hunk 3 above)
Comment (noob247365) [reply src/utils/sidecar.js]: Should be resolved with latest commit

File: src/utils/configuration.js (hunk 2 above)
Comment (noob247365) [reply src/utils/configuration.js]: Should be resolved with latest commit

**Review by danwillm — DISMISSED** [review]

**Pull request closed without merging**